design-documents/module-split.md

```diff
@@ -0,0 +1,83 @@
+# Module split
+
+## About
+
+This document clarifies the following two questions of the original [service isolation proposal](service-isolation.md)
+1. What new modules should be called
+2. What to do with module namespaces, keep old or change so they match new module name
+
+## Design
+
+1. Module naming
+
+    The following is proposed naming for modules:
+    API (service contracts)
+    * CatalogApi _recommended_
+    
+    The following is proposed naming for modules:
+    Admin Ui and admin related business logic (blocks, controllers, static resources, models and resources potentially)
+    * CatalogAdmin
+    * CatalogAdminUi _recommended_
```

Comment:
Should we drop Ui postfix? We can have a case when resource model is only relevant only in storefront, as it's not UI it can be confusing.

Reply:
Decided to leave Ui postfix.